.github/workflows/ci.yml

```diff
@@ -0,0 +1,54 @@
+name: CI
+
+# Trigger the workflow on push or pull request, but only for the master branch
+on:
+  pull_request:
+  push:
+    branches: ["master"]
+
+jobs:
+  generateMatrix:
+    name: "Generate matrix from cabal"
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.set-matrix.outputs.matrix }}
+    steps:
+      - name: Checkout base repo
+        uses: actions/checkout@v3
+      - name: Extract the tested GHC versions
+        id: set-matrix
+        run: |
+          wget https://github.com/Kleidukos/get-tested/releases/download/v0.1.5.0/get-tested-0.1.5.0-linux-amd64 -O get-tested
+          chmod +x get-tested
+          ./get-tested --ubuntu --macos *.cabal >> $GITHUB_OUTPUT
+  tests:
+    name: ${{ matrix.ghc }} on ${{ matrix.os }}
+    needs: generateMatrix
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix: ${{ fromJSON(needs.generateMatrix.outputs.matrix) }}
+    steps:
+      - name: Checkout base repo
+        uses: actions/checkout@v3
+      - name: "Configure ghcup"
+        run: |
+          ghcup config add-release-channel https://raw.githubusercontent.com/haskell/ghcup-metadata/master/ghcup-prereleases-0.0.7.yaml
+      - name: Set up Haskell
+        id: setup-haskell
+        uses: haskell/actions/setup@v2
```

Comment:
Please switch to `haskell-actions/setup@v2`.

Reply:
Ah lovely, thank you.